cpp/include/cuspatial/cuspatial.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+namespace cuSpatial
+{
+	struct Location3D
+	{
+		double lat;
+		double lon;
+		double alt;
+	};
+
+	struct Coord2D
+	{
+		double x;
+		double y;
+	};
+
+	struct TimeStamp
```

Comment:
Can you just use cuDF timestamps? This is a very nonstandard representation of a timestamp -- typically it's an offset from the Epoch.  Do you need to define this in cuSpatial?

Reply:
I looked into cuDF timestamp definition and APIs - they are very complex.  Day of year and day of week etc., which are popular in OLAP type of operations, need to be computed in real time which may take quite some overhead. I also need to compute duration between two timestamps in CUDA kernel and it is unclear to me how to call cuDF APIs for this purpose. We can talk more about this and make changes in the next release.